### DISCUSSION TITLE
Help with hub on RIA please

### DISCUSSION THREAD

**Discussion by n9jcv** [picocomputer/community#68] [Q&A]
Sorry I am a newbie here.  I searched and could not find my issue.  

I just built the rp6502.  I only have the vga and ria installed, no chips.  I get the display on a vga monitor.

I have 2 OTG connectors/cables.  I tried both and when i plug in one end into the RIA and the usb keyboard in the other end the keyboard works.  That tells me my OTGs are good.  Maybe I am wrong?

What does not work is if I plug a hub into the OTG and then a usb keyboard into the hub.  The keyboard does then not work. I have 3 different hubs.  2 not powered and one powered.  Keyboard does not work in any.  What am I missing please?

Hubs

![17039987616868688192369287910478](https://github.com/picocomputer/community/assets/6472203/e46863e0-e633-42bb-9758-9c9e1ccc5598)

Otg working directly. White one.  Black also works.
![17039988311008042195052794413683](https://github.com/picocomputer/community/assets/6472203/7eea6408-969c-4049-97a9-4f7a672c157e)

**Comment by n9jcv** [comment]
As a further test i took this black hub and plugged it into a laptop.  Then i put two usb drives in it.  They are both available in windows.  I dont understand why the keyboard does not work with this hub on the rp6502

![17040003752781558439369459395596](https://github.com/picocomputer/community/assets/6472203/7b699300-1659-42c8-8e87-2ab27fcb9c98)

**Comment by rumbledethumps** [comment]
What keyboard? Did you try a different one?

**Reply by n9jcv** [reply]
I have tried 2 keyboards.  Same result, both work going in otg.  I also just tried swapping the firmwares on the picos, so the vga is now the ria and the ria is the vga.  Same result.

![17040012709911053322751453466981](https://github.com/picocomputer/community/assets/6472203/b20f8643-d993-49d9-a5be-3ce3347c426d)




Both keyboards have a few leds for caps, etc.  Both light up direct in otg but not when plugged in hub, thus i think why it is not workingg. But I can not say why.
 And this is the hub I just proved 2 usb sticks worked on the laptop. Hmmm

And the hub lights up showing me there is power.


![17040013712328939295941418802864](https://github.com/picocomputer/community/assets/6472203/9e62422e-fca5-4b97-9e73-d64d4dddb3f3)

**Reply by rumbledethumps** [reply]
Even without a keyboard, the STATUS command can be sent over USB serial via the VGA USB. These type of problems are in the USB stack which is maintained by someone else. While I'm not much help, the STATUS command may be.

**Reply by n9jcv** [reply]
I dont think I am equipped to debug this myself.  I think you are saying, i could hook the vga pico up to a laptop.  Open arduino or vs code and issue status in serial monitor??  I dont even have vscode installed and not sure i could get all that working. If arduino can do it, I might have a chance with that ????

**Reply by n9jcv** [reply]
Well, I KIND of solved it.  I tried a 3rd keyboard and a different hub and it now works.  

I am think there may be a bad piece of hardware that is non compliant or maybe a bug some where in code, but that is beyond my abilities at this point.  

Here is a pic of a working setup.
Black otg cable to a small 4 port hub.  Zippy usb keyboard in the hub.  

The first 2 keyboards that did not work were dells, so maybe a common factor?  

![17040036443724451760548465415334](https://github.com/picocomputer/community/assets/6472203/930f200c-ef34-4282-a52e-97db11fccfa8)

Also, I just tested with the small white otg and it works as well.

Thanks for the insight @rumbledethumps
